Src/java/cqf-fhir-npm/src/main/java/org/cqframework/fhir/npm/NpmPackageManager.java

```diff
@@ -1,254 +1,127 @@
-                return json.get("url").getAsString();
-            }
-        }
-        return null;
-    }
-
-    private NpmPackage resolveDependency(String canonical, String packageId, String igver) throws IOException {
-        if (packageId != null)
-            return pcm.loadPackage(packageId, igver);
-
-        JsonObject pl;
-        logDebugMessage(IWorkerContext.ILoggingService.LogCategory.INIT, "Fetch Package history from "+Utilities.pathURL(canonical, "package-list.json"));
-        try {
-            pl = fetchJson(Utilities.pathURL(canonical, "package-list.json"));
-        } catch (Exception e) {
-            return null;
-        }
-        if (!canonical.equals(pl.get("canonical").getAsString()))
-            throw new IllegalArgumentException("Canonical mismatch fetching package list for "+canonical+"#"+igver+", package-list.json says "+pl.get("canonical"));
-        for (JsonElement e : pl.getAsJsonArray("list")) {
-            JsonObject o = (JsonObject) e;
-            if (igver.equals(o.get("version").getAsString())) {
-                InputStream src = fetchFromSource(pl.get("package-id").getAsString()+"-"+igver, Utilities.pathURL(o.get("path").getAsString(), "package.tgz"));
-                return pcm.addPackageToCache(pl.get("package-id").getAsString(), igver, src, Utilities.pathURL(o.get("path").getAsString(), "package.tgz"));
-            }
-        }
-        return null;
-    }
-
-    private JsonObject fetchJson(String source) throws IOException {
-        URL url = new URL(source+"?nocache=" + System.currentTimeMillis());
-        HttpURLConnection c = (HttpURLConnection) url.openConnection();
-        c.setInstanceFollowRedirects(true);
-        return JsonTrackingParser.parseJson(c.getInputStream());
-    }
-
-    private InputStream fetchFromSource(String id, String source) throws IOException {
-        logDebugMessage(IWorkerContext.ILoggingService.LogCategory.INIT, "Fetch "+id+" package from "+source);
-        URL url = new URL(source+"?nocache=" + System.currentTimeMillis());
-        URLConnection c = url.openConnection();
-        return c.getInputStream();
-    }
-
-    @Override
-    public void logMessage(String msg) {
-        // TODO: Implement logging here
-        //System.out.println(msg);
-    }
-
-    @Override
-    public void logDebugMessage(IWorkerContext.ILoggingService.LogCategory category, String msg) {
-        logMessage(msg);
-    }
+   private static final Logger logger = LoggerFactory.getLogger(NpmPackageManager.class);
+   private final ImplementationGuide sourceIg;
+   private final FilesystemPackageCacheManager fspcm;
+   private final List<NpmPackage> npmList;
+
+   public NpmPackageManager(ImplementationGuide sourceIg) {
+      this.sourceIg = sourceIg;
+
+      try {
+         // userMode indicates whether the packageCache is within the working directory or in the user home
+         fspcm = new FilesystemPackageCacheManager(true, ToolsVersion.TOOLS_VERSION);
```

Comment:
Can I ask you to go an extra mile and add a way to inject `FilesystemPackageCacheManager` (but keep the existing manager as the default) ? 
This will solve two problems: 
- allow to provide a fake package manager to have better tests (like the test that verifies that dependencies were actually loaded, similar to what i did [here](https://github.com/cqframework/clinical_quality_language/commit/1fbae0f8f0979fb7fcf7f3188a0fe1812b4bd492) ) 
- use the same class with more android friendly version FilesystemPackageCacheManager

Reply:
Applied in latest commit